include/cutlass/arch/arch.h

```diff
@@ -47,20 +47,24 @@ namespace arch {
 CUTLASS_DEVICE
 int LaneId() {
   int ret;
-#if !defined(CUTLASS_ENABLE_SYCL) || defined(__SYCL_CUDA_ARCH__)
+#if defined(__CUDA_ARCH__) || defined(__SYCL_CUDA_ARCH__)
   asm ("mov.u32 %0, %%laneid;" : "=r"(ret) : );
-#endif
   return ret;
+#else
+  CUTE_GCC_UNREACHABLE;
+#endif
 }
 
 /// Computes SM number the thread is running on
 CUTLASS_DEVICE
 int SmId() {
   int ret;
-#if !defined(CUTLASS_ENABLE_SYCL) || defined(__SYCL_CUDA_ARCH__)
+#if defined(__CUDA_ARCH__) || defined(__SYCL_CUDA_ARCH__)
   asm ("mov.u32 %0, %%smid;" : "=r"(ret) : );
-#endif
   return ret;
+#else
+  CUTE_GCC_UNREACHABLE;
```

Comment:
assert?

Reply:
And all other places - I will not comment on each one.

Reply:
Done done done :+1: